### ISSUE TITLE
debugging support in twisted.protocols.socks is inefficient

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7585]
|<img alt="maoy's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| maoy reported|
|-|-|
|Trac ID|trac#285|
|Type|defect|
|Created|2003-09-25 03:54:40Z|

Attachments:

* [bugreport.txt](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/367/367ac64a16d19e2afefcf7c5fab8666dda92f9de/7eae05ee2025ef0d7d93810932290184879b4e19.txt) (39478 bytes) - added by maoy on 2003-09-25 03:54:40Z - 
* [socks_fix.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/367/367ac64a16d19e2afefcf7c5fab8666dda92f9de/f535fb70b5fcfdb3e7670d2e21dd58fe4c4af3fd.diff) (1873 bytes) - added by z3p on 2003-10-21 06:18:19Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__285 285
type__defect defect
reporter__maoy maoy
priority__lowest lowest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__core core
keywords__ 
time__1064462080000000 1064462080000000
changetime__1467798034568445 1467798034568445
version__ 
owner__maoy maoy
cc__itamarst cc__z3p cc__maoy cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
The proxy eats up tons of CPU in line 141 of socks.py,
apparently.

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
Is there an easy way to reproduce this bug?
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="z3p's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|z3p commented|
|-|-|

```
#!html
<pre>
I think itamar fixed this in CVS?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Nope. Just do .encode("hex") or something. The whole function looks likeit could
be trivially optimized (get rid of repeated getPeer() calls, etc..)
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="z3p's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|z3p commented|
|-|-|

```
#!html
<pre>
I can't reproduce the bug even with the old code, but here's a patch that
reworks that section of the code.  Can someone who does see the bug try it?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
Moving from testing to in-progress
(testing means the code *currently* in CVS should work
and just needs testing)
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="z3p's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|z3p commented|
|-|-|

```
#!html
<pre>
Itamar, do you still see this bug?
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="z3p's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|z3p commented|
|-|-|

```
#!html
<pre>
Closing, since the code has been refactored, no one has complained, and no one
see s this bug anymore.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="maoy's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|maoy commented|
|-|-|

```
#!html
<pre>
I checked the 2.0.0 release and I still found the bug there.
"socks_fix.diff"didn't really be applied.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to maoy|
|-|-|

It seems as though the expensive path is only hit if you explicitly request logging of all traffic that is proxied.  Is there a problem with this code other than the fact that the logging is particularly expensive/inefficient?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun commented|
|-|-|

Seems no one is interested in optimizing the debug codepath through socks.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_14"></a>|@glyph set status to `closed`|
|-|-|

I don't think anyone cares to deal with this.  If someone comes along who is interested in it, please attach a benchmark and test code.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_15"></a>|GitHub commented|
|-|-|

In changeset 51d359c0f86aab35dd309386273084e4bb45e121
```
#!CommitTicketReference repository="" revision="51d359c0f86aab35dd309386273084e4bb45e121"
Merge pull request #285 from twisted/8550-rodrigc-names

Author: rodrigc
Reviewer: Lukasa, glyph
Fixes: #8550

Port twisted.names.resolve and twisted.names.tap to Python 3
```
